src/components/Navigation/Pagination/hooks/usePagination.ts

```diff
@@ -21,10 +20,13 @@ export const usePagination = ({
   page: pageProp,
   siblingCount = 1,
 }: UsePaginationProps): UsePaginationItem[] => {
-  const [page, setPageState] = useState(pageProp === undefined ? defaultPage : pageProp);
+  const [page, setPageState] = useCustomEnsuredControl({
+    value: pageProp,
+    defaultValue: defaultPage,
+  });
 
   const handleClick: UsePaginationProps['onChange'] = (event, value) => {
-    if (!pageProp) {
+    if (!pageProp && value !== 0) {
```

Comment:
Can you explain when it's zero here?

Reply:
ellipsis click is treated as zero

https://github.com/Telegram-Mini-Apps/TelegramUI/blob/6a3882a42892116a6203371b63dbcf4e84ac9042/src/components/Navigation/Pagination/hooks/usePagination.ts#L120

Reply:
![ScreenRecording2024-06-10at18 40 01-ezgif com-video-to-gif-converter](https://github.com/Telegram-Mini-Apps/TelegramUI/assets/52068583/969df610-0c27-48aa-8d67-cfd33afe2b5a)

Reply:
Right thing to do is remove onClick from the ellipse div

Reply:
Are you sure? I think someone might want to add some functionality on ellipsis click like in [antd](https://ant.design/components/pagination).

Reply:
You can't change children of this parent component, so developer outside can't control this buttons